web/src/pages/Environments/EnvironmentForm.tsx

```diff
@@ -0,0 +1,75 @@
+import {Button, Form, FormInstance, Input} from 'antd';
+import {useEffect} from 'react';
+import styled from 'styled-components';
+import RequestDetailsHeadersInput from '../../components/CreateTestPlugins/Rest/steps/RequestDetails/RequestDetailsHeadersInput';
+import {
+  IEnvironment,
+  useCreateEnvironmentMutation,
+  useLazyGetEnvironmentSecretListQuery,
+} from '../../redux/apis/TraceTest.api';
+import {EnvironmentState} from './EnvironmentState';
+
+export const Footer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  gap: 8px;
+`;
+
+interface IProps {
+  form: FormInstance<IEnvironment>;
+  state: EnvironmentState;
+  onCancel: () => void;
+}
+
+export const EnvironmentForm: React.FC<IProps> = ({state, form, onCancel}) => {
+  const [createEnvironment] = useCreateEnvironmentMutation();
+  const [loadResultList, {data}] = useLazyGetEnvironmentSecretListQuery();
+  useEffect(() => {
+    if (state.environment) loadResultList({environmentId: state.environment?.id || ''});
```

Comment:
Are we expecting to get the environment data without the entries list?

Reply:
the environment initialState is undefined.